src/types/ReferenceType.ts

```diff
@@ -487,8 +487,6 @@ export class BinaryOperatorReferenceType extends BscType {
  * Use this class for when there is a presumable array type that is a referenceType
  */
 export class ArrayDefaultTypeReferenceType extends BscType {
-    cachedType: BscType;
```

Comment:
remove caching here - just always do the lookup, and rely on proper caching upstream